src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinStr = '';
+    const separatorStr = `${separator}`;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] !== null && this[i] !== undefined) {
+        joinStr += this[i];
+      } else if (this[i] !== null && this[i] !== undefined) {
+        joinStr += this[i] + separatorStr;
+      } else if (i === this.length - 1) {
+        break;
+      } else {
+        joinStr += separatorStr;
+      }
```

Comment:
Too complicated logic, need to be refactored

Reply:
Just add element to result string if it's not a `null` and `undefined`. Same with separator, just add it if element is not last